File: src/main/java/visitor/user/Student.java
package visitor.user;

import visitor.visitor.Visitor;

/** @author Elias (siran0611@gmail.com) */
public class Student extends User {
  public Student(String name, String identity, String clazz) {
    super(name, identity, clazz);
  }

  @Override
  public void accept(Visitor visitor) {
    visitor.visit(this);
  }

  public int ranking() {
    return (int) (Math.random() * 100);
  }
}

File: src/rust/visitor/user/student.rs
use crate::rust::visitor::user::{User, UserEnum};
use crate::rust::visitor::visitor::Visitor;

#[derive(Debug, Clone)]
pub struct Student {
    pub name: &'static str,
    identity: &'static str,
    class: &'static str,
}

impl User for Student {
    fn accept(&self, visitor: &impl Visitor) {
        let student = UserEnum::Student(self.clone());
        visitor.visit(student);
    }
}

impl Student {
    pub fn new(name: &'static str, identity: &'static str, class: &'static str) -> Self {
        Self { name, identity, class }
    }

    pub fn ranking(&self) -> i32 {
        120
    }
}